patches/api/0168-BlockDestroyEvent.patch

```diff
@@ -36,23 +37,42 @@ index 0000000000000000000000000000000000000000..051b2ef76a914228338fa28553ad739b
 + * Events such as leaves decaying, pistons retracting (where the block is moving), does NOT fire this event.
 + *
 + */
-+public class BlockDestroyEvent extends BlockEvent implements Cancellable {
++public class BlockDestroyEvent extends BlockExpEvent implements Cancellable {
 +
 +    private static final HandlerList handlers = new HandlerList();
 +
 +    @NotNull private final BlockData newState;
 +    private boolean willDrop;
 +    private boolean playEffect = true;
++    private BlockData effectBlock;
 +
 +    private boolean cancelled = false;
 +
 +    public BlockDestroyEvent(@NotNull Block block, @NotNull BlockData newState, boolean willDrop) {
-+        super(block);
++        super(block, 0);
 +        this.newState = newState;
 +        this.willDrop = willDrop;
 +    }
 +
 +    /**
++     * Get the effect that will be played when the block is broken.
++     * @return block break effect
++     */
++    @NotNull
++    public BlockData getEffectBlock() {
++        return this.effectBlock;
++    }
++
++    /**
++     * Sets the effect that will be played when the block is broken.
++     *
++     * @param effectBlock block effect
++     */
```

Comment:
Should mention/link the `#playEffect()` as controlling if the effect plays at all and
to disable effect, use `#setPlayEffect()`